libs/wingsdk/src/cloud/bucket.ts

```diff
@@ -19,20 +19,30 @@ export interface BucketProps {
 }
 
 /**
- * Functionality shared between all `Bucket` implementations.
+ * Represents a cloud object store.
+ *
+ * @inflight `@winglang/sdk.cloud.IBucketClient`
  */
-export abstract class BucketBase extends Resource {
+export abstract class Bucket extends Resource {
+  /**
+   * Create a new bucket.
+   */
+  public static newBucket(
```

Comment:
It's a good practice to give static factory methods explicit names because if you extend the class, the name becomes ambigious:

```js
class MyBucket extends Bucket { }
```

And now:

```js
const b = MyBucket.new(...); // returns `Bucket`
```

Per @Chriscbr's suggestion I am going to make these internal, so we can tweak this later.